### PR TITLE
Adjust to recent stackmap call position change.

### PR DESCRIPTION
**Pull request by ptersilie**
Stackmap calls now appear after (not before) a call, so we need to adjust our code accordingly.

Companion PR: https://github.com/ykjit/ykllvm/pull/52

### CODE REVIEW

File: ykfr/src/lib.rs [hunk 1]
```diff
@@ -70,7 +69,11 @@ impl Frame {
 
 fn get_stackmap_call(pc: Value) -> Value {
     debug_assert!(pc.is_instruction());
-    let sm = unsafe { Value::new(LLVMGetPreviousInstruction(pc.get())) };
+    let sm = if pc.is_call() {
```
Comment (vext01) [inline ykfr/src/lib.rs:72]: One case for branches, one case for calls? SHould we add some comments?

File: ykfr/src/lib.rs [hunk 2]
```diff
@@ -115,27 +118,13 @@ impl FrameReconstructor {
...
         let file = fs::File::open(&pathb.as_path()).unwrap();
         let exemmap = unsafe { memmap2::Mmap::map(&file).unwrap() };
         let object = object::File::parse(&*exemmap).unwrap();
         let sec = object.section_by_name(".llvm_stackmaps").unwrap();
-        let text = object.section_by_name(".text").unwrap();
-
-        // Create a capstone dissassembler so we can figure out the size of the instruction we
-        // return to in order to adjust the return address in our reconstructed stackmap.
-        // FIXME: Find a way to move this into AOT (e.g. maybe add this information to stackmaps).
-        let cs = Capstone::new()
```
Comment (vext01) [inline ykfr/src/lib.rs:132]: fantastic that this is going away.

File: ykfr/src/lib.rs (hunk 1 above)
Comment (ptersilie) [reply ykfr/src/lib.rs:74]: Try [55ccf18](https://github.com/ykjit/yk/pull/589/commits/55ccf187578d930b3945d6f6b2e7fde3e9d455c7)

Comment (vext01) [reply ykfr/src/lib.rs:74]: perfect.

**Comment by vext01** [comment]
Please squash.

File: ykfr/src/lib.rs [hunk 3]
```diff
@@ -303,22 +294,10 @@ impl FrameReconstructor {
...
-                            }
-                            8 => {
-                                let dval = unsafe { ptr::read(val as *mut u64) };
-                                unsafe { ptr::write(temp as *mut u64, dval) };
-                            }
-                            _ => {
-                                todo!()
-                            }
-                        }
+                        unsafe { libc::memcpy(temp, val as *const c_void, size as usize) };
```
Comment (ltratt) [inline ykfr/src/lib.rs:297]: `as` in Rust is dangerous, so we should always add a (brief) comment explaining why it's safe. In this case I'm not sure what type `size` has before the cast.

Comment (ptersilie) [reply ykfr/src/lib.rs]: `size` is a `u64` (`c_ulonglong`). But I've changed this to use `try_from` in [607e912](https://github.com/ykjit/yk/pull/589/commits/607e9126124fbc543121b222fa75ad0899bc212f)

**Comment by ltratt** [comment]
Please squash.

**Comment by ptersilie** [comment]
Squashed!

**Comment by ltratt** [comment]
Weird. That passed earlier.

bors try

**Comment by ltratt** [comment]
Let's see if 2023 fixes things.

bors try

**Comment by ptersilie** [comment]
Oh I think I have an idea what's going wrong here.

**Comment by ptersilie** [comment]
This should fix things. Let's try again.

**Comment by ptersilie** [comment]
Needs squashing though.

**Comment by ltratt** [comment]
bors try

If bors succeeds, then please squash.

**Comment by ptersilie** [comment]
Squashed.

**Pull request merged**